openmicroscopy/openmicroscopy
# search by annotation type

# What this PR does

See https://trello.com/c/HBZKcjw8/42-search-by-specific-annotations
This allows users to search by "File" and "Tag" annotations, not just by All annotations.

# Testing this PR

1. Find or create projects, datasets, images etc that have the same keyword in name OR Comment OR tag OR file content (e.g. add image to figure, add keyword in label and export to PDF to attach PDF to image).

 2. Search for keyword. By default you should find results with keyword in Name, Tag and File.

 3. Check the "Annotations" filter checkbox. Radio buttons should be shown, with "All annotations" checked by default. This should find results with Comment, Tag and File.

4. Check the Tag or File radio buttons. Now should only find results with Tag or File respectively.

5. Uncheck the "Annotations" checkbox to hide the radio buttons. Now we should get results for Name and Annotations as before.

![screen shot 2018-01-22 at 10 55 01](https://user-images.githubusercontent.com/900055/35218847-108d2fb4-ff68-11e7-8a7d-a05f3d1710dc.png)
